synapseclient/client.py

```diff
@@ -654,7 +654,7 @@ def getUserProfile(
             )
         )
 
-    def _findPrincipals(self, query_string):
+    def _findPrincipals(self, query_string: str) -> list:
```

Comment:
To be more specific I think this would be what we want:

```suggestion
    def _findPrincipals(self, query_string: str) -> typing.List[UserGroupHeader]:
```

Reply:
Made this change in my latest commit